aptos-move/aptos-intent/src/codegen.rs

```diff
@@ -0,0 +1,489 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    ArgumentOperation, BatchArgument, BatchedFunctionCall, PreviousResult, APTOS_INTENT_KEY,
+};
+use move_binary_format::{
+    access::{ModuleAccess, ScriptAccess},
+    errors::{PartialVMError, PartialVMResult},
+    file_format::*,
+};
+use move_core_types::{
+    account_address::AccountAddress,
+    identifier::{IdentStr, Identifier},
+    language_storage::{ModuleId, TypeTag},
+    metadata::Metadata,
+    transaction_argument::TransactionArgument,
+    vm_status::StatusCode,
+};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+#[derive(Default)]
```

Comment:
If we want to make this a general-use library, then I feel like we'll need to come up with a way to make it work with an existing non-empty CompiledScript/Module.

Reply:
Ok now I see your point. Basically you want to get rid of the intent codegen logic and move the logic to the builder, so that what's remaining would be purely reusable code for manipulating compiled script.